paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/procedure/CompactProcedure.java

```diff
@@ -251,6 +274,15 @@ private boolean execute(
                                 .getOrElse(null);
         PartitionPredicate partitionPredicate =
                 PartitionPredicate.fromPredicate(partitionType, filter);
+
+        if (clusterIncrementalEnabled) {
```

Comment:
Should return after `boolean clusterIncrementalEnabled = table.coreOptions().clusteringIncrementalEnabled();`.